File: 0010/P0010.hs
{-# LANGUAGE UnicodeSyntax #-}

readInt ∷ IO Int
readInt = readLn

main ∷ IO ()
main = do
  t <- readInt
  countNthPrime t [3, 2] [2, 3]

countNthPrime ∷ Int → [Int] → [Int] → IO ()
countNthPrime 0 _ _     = return ()
countNthPrime i primes revprimes = do
  n <- readInt
  let (primes', revprimes') = prime n (head primes + 2) primes revprimes
  print $ sum $ filter (<=n) primes'
  countNthPrime (i - 1) primes' revprimes'

prime ∷ Int → Int → [Int] → [Int] → ([Int], [Int])
prime n nextPrime primes revprimes
  | head primes >= n                   = (primes, revprimes)
  | isPrime nextPrime primes revprimes = prime n (nextPrime + 2) (nextPrime : primes) (revprimes ++ [nextPrime])
  | otherwise                          = prime n (nextPrime + 2) primes revprimes

isPrime ∷ Int → [Int] → [Int] → Bool
isPrime n primes revprimes = isPrime' n (s n) revprimes
  where
    s = floor . sqrt . fromIntegral
    isPrime' n s primes
      | null primes              = True
      | head primes > s          = True
      | n `mod` head primes == 0 = False
      | otherwise                = isPrime' n s (tail primes)

File: 0010/P0010.py
from math import sqrt

def is_prime(n, primes):
  s = int(sqrt(n))
  for prime in primes:
    if n % prime == 0:
      return False
    if prime >= s:
      return True
  return True

def calculate_primes(n, primes):
  next_prime = primes[-1] + 2
  while primes[-1] <= n:
    if is_prime(next_prime, primes):
      primes.append(next_prime)
    next_prime += 2
  return primes

if __name__ == "__main__":
  primes = [2, 3]
  t = int(input())
  for i in range(t):
    n = int(input())
    primes = calculate_primes(n, primes)
    result = sum(prime for prime in primes if prime <= n)
    print(result)